### ISSUE TITLE
zero-sized files in tarballs do not call `end` handler

### ISSUE THREAD

**Issue by mattgodbolt** [mafintosh/tar-stream#145]
If you are extracting an archive with zero-sized files in it, then the stream your `entry` handler is passed has already been `end`-ed. This means the `on('end')` event is not triggered.

Our code looks something like:
```ts
extract.on('entry', async (header, stream, next) => {
  //...
    const filestream = fs.createWriteStream(filepath);
    stream
        .on('error', error => {
            logger.error(`Error in stream handling: ${error}`);
            reject(error);
        })
        .on('end', () => {
            next();  /// <--- this handler is never called for zero-sized files, hanging the stream forever
        })
        .pipe(filestream);
    stream.resume();
    if (header.size === 0) next(); //// <--- adding this line "fixes" the issue
```

**Comment by mattgodbolt** [comment]
The stream for empty files is created here: https://github.com/mafintosh/tar-stream/blob/ca0e270f11dc61f507f7972e53071b8fa99e66bf/extract.js#L174

and the stream has already been ended here: https://github.com/mafintosh/tar-stream/blob/ca0e270f11dc61f507f7972e53071b8fa99e66bf/extract.js#L12

so there's no `end()` event that's emitted _after_ we have had a change to register an on end handler (as best I can tell).

**Comment by mafintosh** [comment]
Fixed in 3.1.0
